src/Compilers/Core/Portable/Symbols/TypedConstant.cs

```diff
@@ -94,32 +97,35 @@ public ImmutableArray<TypedConstant> Values
                     return default(ImmutableArray<TypedConstant>);
                 }
 
-                return (ImmutableArray<TypedConstant>)_value;
+                return (ImmutableArray<TypedConstant>)_value!;
             }
         }
 
+        [return: MaybeNull]
         internal T DecodeValue<T>(SpecialType specialType)
         {
+#pragma warning disable CS8717 // A member returning a [MaybeNull] value introduces a null value when 'T' is a non-nullable reference type.
             TryDecodeValue(specialType, out T value);
+#pragma warning restore CS8717 // A member returning a [MaybeNull] value introduces a null value when 'T' is a non-nullable reference type.
```

Comment:
What motivates the usage of `pragma warning disable` versus `!`?

Reply:
Both approaches have merits. This one was chosen as a means to highlight the negative impact this warning is having on the overall NRT development experience.

Reply:
Imo using `!` isn't right, because we're not "suppressing" nulls, we just pass it to the caller. So I think the compiler shouldn't warn in the first place.